src/views/Project/Details/Connector/ConnectorDetails/index.tsx

```diff
@@ -25,37 +29,52 @@ interface OwnProps {
     className?: string;
     projectId: number;
     details: Connector;
+    onConnectorDelete: (id: number) => void;
+    onConnectorEdit: (connector: Connector) => void;
 }
 
-const sourceKeySelector = (source: ConnectorSource) => source.key;
+const sourceKeySelector = (source: UnifiedConnectorSource) => source.source;
 
 function ProjectConnectorDetail(props: OwnProps) {
     const {
+        onConnectorDelete,
+        onConnectorEdit,
         projectId,
         className,
         details,
     } = props;
 
     const {
+        id: connectorId,
         title,
         updatedOn,
         disabled,
         sources,
     } = details;
 
+    const [connectorToBeDeleted, setConnectorToBeDeleted] = useState<number | undefined>();
+
+    const [pendingConnectorDelete,,, deleteConnectorTrigger] = useRequest({
+        url: `server://projects/${projectId}/unified-connectors/${connectorToBeDeleted}/`,
+        method: 'DELETE',
+        onSuccess: () => {
+            onConnectorDelete(connectorId);
```

Comment:
Are we sure thsi connectorId doesn't change? Why not use `connectorToBeDeleted` here?